tools/autograd/gen_variable_type.py

```diff
@@ -493,6 +493,8 @@
     # These functions are expected to change impl or storage of input tensors
     "set_",
     "_cudnn_rnn_flatten_weight",
+    # Nested Tensors related functions
+    "_nested_tensor_size",
```

Comment:
nt.to_padded_tensor also throws a use_count and storage_count error in debug. In the nested_tensor_size case we take in a tensor T and return a tensor R who's impl points to the same as T.nested_size and whose storage pointer is also pointing T.nested_size.storage.

This function should never really be used in context of autograd.. hopefully.
However to_padded_tensor certainly will - the next PR will add the backward formula. 

Question: if I add  `to_padded_tensor` to this list, is this going to lead to some insidious bugs that are hard to find.  Also would it be better to have a to_padded_tensor and to_padded_tensor_copy. Not entirely sure if this PR impacts this at all:  #79616.  But seeing as this list is so small I am trying to figure out if Nested Tensor just doesn't fit into the current paradigm or we should be hesitant appending to it.

Reply:
Storage use count check is useful for enforcing that a view relationship is properly established when the output shares storage (StorageImpl use count > 1) - this makes sure the grad_fn of the base is properly updated when in-placing on a view. Checking TensorImpl use count == 1 is important to prevent autograd from overwriting existing grad_fn with set_history.

`_nested_tensor_size` size should be OK because we don't use it with autograd, but yeah we probably want to do something else for `to_padded_tensor`.

Reply:
Yup you are totally right about the list, will add the comment